doc/developers/contributing.rst

```diff
@@ -207,6 +207,167 @@ and Cython optimizations.
    <http://astropy.readthedocs.org/en/latest/development/workflow/development_workflow.html>`_
    sections.
 
+Rebasing and squashing commits
+------------------------------
+
+Squashing commits
+^^^^^^^^^^^^^^^^^
+
+To keep the history clean and to help backtrack regressions to a single commit we
+prefer less (but meaningful) number of commits.
```

Comment:
umm I'm not quite sure about this.... It still retains the same meaning but makes it unnecessarily verbose. What do you think?